x-pack/plugins/apm/server/lib/services/get_services/get_services_items_stats.ts

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { arrayUnionToCallable } from '../../../../common/utils/array_union_to_callable';
+import {
+  PROCESSOR_EVENT,
+  TRANSACTION_DURATION,
+  AGENT_NAME,
+  SERVICE_ENVIRONMENT,
+} from '../../../../common/elasticsearch_fieldnames';
+import { mergeProjection } from '../../../../common/projections/util/merge_projection';
+import { ProcessorEvent } from '../../../../common/processor_event';
+import {
+  ServicesItemsSetup,
+  ServicesItemsProjection,
+} from './get_services_items';
+
+const MAX_NUMBER_OF_SERVICES = 500;
+
+const getDeltaAsMinutes = (setup: ServicesItemsSetup) =>
+  (setup.end - setup.start) / 1000 / 60;
+
+interface AggregationParams {
+  setup: ServicesItemsSetup;
+  projection: ServicesItemsProjection;
+}
+
+export const getTransactionDurationAverages = async ({
+  setup,
+  projection,
+}: AggregationParams) => {
+  const { client, indices } = setup;
+
+  const response = await client.search(
+    mergeProjection(projection, {
+      size: 0,
+      index: indices['apm_oss.transactionIndices'],
```

Comment:
A comment more on projections: In this case I don't know how to see that the projection won't override the `index`.

Reply:
Maybe I'm misunderstanding - but this is a merge, so values are applied from left to right. ie, `indices['apm_oss.transactionIndices']` will override whatever the projection has defined.

Reply:
Okay, that makes good sense